temba/classifiers/types/dialogflow/train_bot.py

```diff
@@ -1,148 +1,164 @@
+import base64
 import csv
 import logging
+import pickle
+from typing import Union, Any
 
 from google.cloud import dialogflow_v2
-
+from google.api_core import exceptions
+from google.cloud.dialogflow_v2.services.intents import pagers
 
 logger = logging.getLogger(__name__)
 
 
 class TrainingClient:
-    def __init__(self, csv_data: list, languages: list, credential: dict):
+    def __init__(self, csv_data: list, languages: list, credential: dict, messages: dict = None) -> None:
         self.credential = credential
         self.csv_data = csv_data
         self.languages = languages
-        fill_list = [[]] * len(languages)
-        fill_zero = [0] * len(languages)
-        errors_per_lang = dict(zip(languages, fill_list))
-        count_per_lang = dict(zip(languages, fill_zero))
-        self.messages = dict(
-            errors=[],
-            update_errors=errors_per_lang.copy(),
-            create_errors=errors_per_lang.copy(),
-            created=count_per_lang.copy(),
-            updated=count_per_lang.copy(),
-        )
+        self.intents_requests = []
+
+        if messages and len(messages) > 0:
+            self.messages = messages
+        else:
+            fill_list = [[]] * len(languages)
+            fill_zero = [0] * len(languages)
+            errors_per_lang = dict(zip(languages, fill_list))
+            count_per_lang = dict(zip(languages, fill_zero))
+            self.messages = dict(
+                errors=[],
+                update_errors=errors_per_lang.copy(),
+                create_errors=errors_per_lang.copy(),
+                created=count_per_lang.copy(),
+                updated=count_per_lang.copy(),
+            )
+
+    def merge_intents(self, csv_reader: csv.DictReader, language_code: str) -> list:
+        lang_headers = self.get_language_headers(language_code)
+        grouped_by_intents = {}
+        for row in csv_reader:
+            intent_header = lang_headers["intent"]
+            question_header = lang_headers["training_phrase"]
+            answer_header = lang_headers["answer"]
+            try:
+                intent_name = row.get(intent_header)
+            except KeyError:
```

Comment:
I don't think `KeyError` is an exception from `.get()`, here you can implement this way:
```
intent_name = row.get(intent_header, row.get("intents"))
```